test/facility-loan/put-facility-loan-update-expiry-date.api-test.ts

```diff
@@ -0,0 +1,134 @@
+    requestToGetLoanInAcbs().reply(200, acbsGetExistingLoanResponse);
+  };
+
+  const givenRequestToGetLoanInAcbsSucceeds = () => {
+    givenRequestToGetLoanInAcbsSucceedsWithResponse(acbsGetExistingLoanResponse);
+  };
+
+  const requestToUpdateLoanInAcbs = () =>
+    nock(ENVIRONMENT_VARIABLES.ACBS_BASE_URL).put(`/Portfolio/${portfolioIdentifier}/Loan/${loanIdentifier}`).matchHeader('authorization', `Bearer ${idToken}`);
+
+  const givenAcbsUpdateLoanSucceedsWithResponse = (acbsUpdateLoanRequest: AcbsUpdateLoanRequest) => {
+    requestToUpdateLoanInAcbs().reply(200, acbsUpdateLoanRequest);
+  };
+
+  const givenAcbsUpdateLoanSucceeds = () => {
+    givenAcbsUpdateLoanSucceedsWithResponse(acbsUpdateLoanRequest);
+  };
+
+  const { idToken, givenAuthenticationWithTheIdpSucceeds } = withAcbsAuthenticationApiTests({
+    givenRequestWouldOtherwiseSucceed: () => {
+      givenRequestToGetLoanInAcbsSucceeds();
+      givenAcbsUpdateLoanSucceeds();
+    },
+    makeRequest: () => makeRequest(updateLoanExpiryDateRequest),
+    successStatusCode: 200,
+  });
+
+  withClientAuthenticationTests({
+    givenTheRequestWouldOtherwiseSucceed: () => {
+      givenAuthenticationWithTheIdpSucceeds();
+      givenRequestToGetLoanInAcbsSucceeds();
+      givenAcbsUpdateLoanSucceeds();
+    },
+    makeRequestWithoutAuth: (incorrectAuth?: IncorrectAuthArg) =>
+      api.patchWithoutAuth(
+        getUpdateExpiryDateUrl({ facilityIdentifier, loanIdentifier }),
+        updateLoanExpiryDateRequest,
+        incorrectAuth?.headerName,
+        incorrectAuth?.headerValue,
+      ),
+  });
+
+  describe('field tests', () => {
+    withDateOnlyFieldValidationApiTests({
+      fieldName: 'expiryDate',
+      validRequestBody: updateLoanExpiryDateRequest,
+      makeRequest: (body: unknown) => makeRequest(body),
+      givenAnyRequestBodyWouldSucceed: () => {
+        givenAuthenticationWithTheIdpSucceeds();
+        givenRequestToGetLoanInAcbsSucceeds();
+        givenAcbsUpdateLoanSucceeds();
+      },
+    });
+  });
+
+  describe('URL parameter validation', () => {
+    withFacilityIdentifierUrlValidationApiTests({
+      makeRequestWithFacilityId: (facilityIdentifier: string) =>
+        api.patch(getUpdateExpiryDateUrl({ facilityIdentifier, loanIdentifier }), updateLoanExpiryDateRequest),
+      givenRequestWouldOtherwiseSucceedForFacilityId: () => {
+        givenAuthenticationWithTheIdpSucceeds();
+        givenRequestToGetLoanInAcbsSucceeds();
+        givenAcbsUpdateLoanSucceeds();
+      },
+      successStatusCode: 200,
+    });
+
+    withLoanIdentifierUrlValidationApiTests({
+      makeRequestWithLoanId: (loanIdentifier: string) => api.patch(getUpdateExpiryDateUrl({ facilityIdentifier, loanIdentifier }), updateLoanExpiryDateRequest),
+      givenRequestWouldOtherwiseSucceedForLoanId: () => {
+        givenAuthenticationWithTheIdpSucceeds();
+        givenRequestToGetLoanInAcbsSucceeds();
+        givenAcbsUpdateLoanSucceeds();
```

Comment:
These 2 mocks are probably going to need to know the loanId being used in the request?

Reply:
So oddly -- updateLoanSucceeds doesn't as we pull it straight from the getLoan response